src/coreclr/src/vm/dynamicmethod.cpp

```diff
@@ -1197,6 +1197,28 @@ LCGMethodResolver::IsValidStringRef(mdToken metaTok)
     return GetStringLiteral(metaTok) != NULL;
 }
 
+LPCWSTR
+LCGMethodResolver::GetStringLiteral(mdToken metaTok, int* length)
+{
+    STANDARD_VM_CONTRACT;
+
+    GCX_COOP();
+
+    STRINGREF str = GetStringLiteral(metaTok);
+    if (str == NULL)
+    {
+        *length = -1;
+        return NULL;
+    }
+    else
+    {
+        *length = str->GetStringLength();
+        SString sstr;
+        str->GetSString(sstr);
+        return sstr.GetUnicode();
```

Comment:
Not sure in this code (for getting the actual string ptr in dynamic context, length is fine).

Reply:
This is use-after-free bug. The string will be released by the on-stack SString destructor.

Reply:
I think you should return NULL from here and make sure that the JIT can deal with it.